test/OpenTelemetry.Exporter.Zipkin.Tests/ZipkinActivityExporterTests.cs

```diff
@@ -0,0 +1,213 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using OpenTelemetry.Exporter.Zipkin.Implementation;
+using OpenTelemetry.Internal.Test;
+using OpenTelemetry.Resources;
+using Xunit;
+
+namespace OpenTelemetry.Exporter.Zipkin.Tests
+{
+    public class ZipkinActivityExporterTests : IDisposable
+    {
+        private static readonly ConcurrentDictionary<Guid, string> Responses = new ConcurrentDictionary<Guid, string>();
+
+        private readonly IDisposable testServer;
+        private readonly string testServerHost;
+        private readonly int testServerPort;
+
+        static ZipkinActivityExporterTests()
+        {
+            Activity.DefaultIdFormat = ActivityIdFormat.W3C;
+            Activity.ForceDefaultIdFormat = true;
+
+            var listener = new ActivityListener
+            {
+                ShouldListenTo = _ => true,
+                GetRequestedDataUsingParentId = (ref ActivityCreationOptions<string> options) => ActivityDataRequest.AllData,
+                GetRequestedDataUsingContext = (ref ActivityCreationOptions<ActivityContext> options) => ActivityDataRequest.AllData,
+            };
+
+            ActivitySource.AddActivityListener(listener);
+        }
+
+        public ZipkinActivityExporterTests()
+        {
+            this.testServer = TestHttpServer.RunServer(
+                ctx => ProcessServerRequest(ctx),
+                out this.testServerHost,
+                out this.testServerPort);
+
+            static void ProcessServerRequest(HttpListenerContext context)
+            {
+                context.Response.StatusCode = 200;
+
+                using StreamReader readStream = new StreamReader(context.Request.InputStream);
+
+                string requestContent = readStream.ReadToEnd();
+
+                Responses.TryAdd(
+                    Guid.Parse(context.Request.QueryString["requestId"]),
+                    requestContent);
+
+                context.Response.OutputStream.Close();
+            }
+        }
+
+        public void Dispose()
+        {
+            this.testServer.Dispose();
+        }
+
+        [Fact]
```

Comment:
If possible, test with UseShortTraceIds = true as well.

Reply:
Added UseShortTraceIds to options and modified it.